src/trio/_tests/test_channel.py

```diff
@@ -625,3 +625,48 @@ async def agen(events: list[str]) -> AsyncGenerator[None]:
                         events.append("body cancel")
                         raise
     assert events == ["body cancel", "agen cancel"]
+
+
+async def test_as_safe_channel_genexit_exception_group() -> None:
+    @as_safe_channel
+    async def agen() -> AsyncGenerator[None]:
+        try:
+            async with trio.open_nursery():
+                yield
+        except BaseException as e:
+            assert isinstance(e, BaseExceptionGroup)  # noqa: PT017  # we reraise
```

Comment:
not that I think it's especially necessary here, but you could do `assert RaisesGroup(GeneratorExit).matches(e)` instead of `isinstance`

Reply:
Unfortunately that also leads to a PT017 warning. I'm somewhat tempted to remove this assert... I'm not sure if that's a good idea though.

Reply:
yeah it wasn't specifically to avoid the PT017, just a stronger check than `isinstance`